### ISSUE TITLE
Hive box(xyz).delete(keyhere) not working

### ISSUE THREAD

**Issue by UmairSaqibBhutta** [hivedb/hive#1170]
I am using hive for data persistence and now I want to delete data at a particular key of a box. But `delete ` method is not working, My code is :
` final box = Hive.box("boxName");
                                    box.delete(keyName);`
but its not working.

**Comment by friedScotch** [comment]
`delete()` is an async function. Did you try awaiting it?

**Comment by LiveRock** [comment]
Was it resolved? Facing same issue even with await
https://github.com/hivedb/hive/issues/1197

**Comment by LiveRock** [comment]
Will Hive store/save with duplicate keys?
Like will it create multiple records or just overwrite the record of the same key?

**Comment by UmairSaqibBhutta** [comment]
It will replace/ Overwrite the previous record with the new one.

**Comment by friedScotch** [comment]
> 

The official doc does not use async, however if you look inside box_base.dart, delete() is an async function

![grafik](https://user-images.githubusercontent.com/99123380/233573708-5307e518-e118-433c-9cb8-e5361cb3e15c.png)

**Comment by darielkurt** [comment]
@LiveRock @UmairSaqibBhutta No, it does not overwrite the value, it only appends. See this [link](https://github.com/isar/hive/issues/417#issuecomment-682247151)
